src/utils/CameraUtils.js

```diff
@@ -0,0 +1,340 @@
+        const animationFrameRequester = () => {
+            tweenGroup.update();
+            this.updateMatrixWorld(true);
+            this.applyTransformToCamera(view, camera);
+            view.notifyChange(camera);
+        };
+
+        function removeAll() {
+            tweenGroup.removeAll();
+            view.removeFrameRequester(MAIN_LOOP_EVENTS.BEFORE_RENDER, animationFrameRequester);
+        }
+
+        animations[0].onComplete(this.deferred.resolve);
+        animations[0].onStop(this.deferred.resolve);
+        animations.forEach(anim => anim.start());
+
+        view.addFrameRequester(MAIN_LOOP_EVENTS.BEFORE_RENDER, animationFrameRequester);
+        view.notifyChange(camera);
+
+        this.deferred.promise.then(() => removeAll());
+
+        return this.deferred;
+    },
+    stop(view) {
+        if (view && this.placeTargetOnGround) {
+            view.removeFrameRequester(MAIN_LOOP_EVENTS.BEFORE_RENDER, this.placeTargetOnGround);
+            this.placeTargetOnGround = null;
+        }
+        if (this.deferred) {
+            this.deferred.resolve();
+            return this.deferred.promise;
+        } else {
+            return Promise.resolve();
+        }
+    },
+    // update target position to coordinate's altitude
+    addPlaceTargetOnGround(view, camera, coord, options = { t: 1.0 }) {
+        if (view && camera) {
+            const startAltitude = this.target.position.z;
+            this.placeTargetOnGround = () => {
+                const result = DEMUtils.getElevationValueAt(tileLayer(view), coord || this.coord, DEMUtils.PRECISE_READ_Z);
+                const altitude = Math.max(0, result ? result.z : 0);
+                this.target.position.z = startAltitude * (1.0 - options.t) + altitude * options.t;
+                this.target.updateMatrixWorld(true);
+                this.applyTransformToCamera(view, camera);
+            };
+            this.placeTargetOnGround();
+            view.addFrameRequester(MAIN_LOOP_EVENTS.BEFORE_RENDER, this.placeTargetOnGround);
+        }
+    },
+});
+
+Object.defineProperties(RigCamera.prototype, {
+    tilt: { get() { return THREE.Math.radToDeg(this.target.rotation.x); } },
+    heading: { get() { return wrapTo180((THREE.Math.radToDeg(this.target.rotation.z) + 180)); } },
+    range: { get() { return this.camera.position.y; } },
+});
+/**
+ * @typedef {Object} cameraTransformOptions
+ * @property {Coordinate} coordinate Camera look at coordinate
+ * @property {Number} tilt camera's tilt
+ * @property {Number} heading camera's heading
+ * @property {Number} range camera distance to target coordinate
+ * @property {Number} time duration of the animation
+ */
+/**
+ * @module CameraUtils
+ */
+export default {
+    /**
+     * Stop camera's animation
+     *
+     * @param      {View}  view    The camera view
+     * @param      {Camera}  camera  The camera to stop animation
+     */
+    stop(view, camera) {
+        getRig(view, camera).stop(view);
```

Comment:
This seems sloppy, I would prefer to see this going in `View`, as it is something really specific to the view. Why not instantiating a `RigCamera` when creating a `View` ? Is it to avoid dependency between those two class ? Maybe putting it inside a `Controls` abstract class would be better ?